docs/pages/hooks/useAddressInput.mdx

```diff
@@ -0,0 +1,194 @@
+| `settledValue`          | `string \| undefined` | The debounced and validated input value (undefined while debouncing).|
+| `debouncedValue`        | `string`              | The debounced input value.                                         |
+
+## Live Examples
+
+### Basic Usage
+
+Simple ENS name to address resolution.
+
+<DocsProvider>
+  <UseAddressInputBasicExample />
+</DocsProvider>
+
+```tsx
+import React, { useState } from "react";
+import { useAddressInput } from "@scaffold-ui/hooks";
+
+function BasicAddressInput() {
+  const [value, setValue] = useState("");
+  const { ensAddress, isEnsAddressLoading } = useAddressInput({ value });
+
+  return (
+    <div>
+      <input
+        value={value}
+        onChange={(e) => setValue(e.target.value)}
+        placeholder="Enter ENS name (e.g., vitalik.eth)"
+      />
+
+      {isEnsAddressLoading && <p>Resolving...</p>}
+
+      {ensAddress && (
+        <p><strong>Address:</strong> {ensAddress}</p>
+      )}
+    </div>
+  );
+}
+```
+
+### Bidirectional Resolution
+
+Resolves both ENS names to addresses and addresses to ENS names, with avatar display.
+
+<DocsProvider>
+  <UseAddressInputWithResolutionExample />
+</DocsProvider>
+
+```tsx
+import React, { useState } from "react";
+import { useAddressInput } from "@scaffold-ui/hooks";
+
+function BidirectionalResolution() {
+  const [value, setValue] = useState("");
+  const {
+    ensAddress,
+    ensName,
+    ensAvatar,
+    isEnsAddressLoading,
+    isEnsNameLoading
+  } = useAddressInput({ value });
+
+  return (
+    <div>
+      <input
+        value={value}
+        onChange={(e) => setValue(e.target.value)}
+        placeholder="Try: vitalik.eth or 0xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045"
+        disabled={isEnsAddressLoading || isEnsNameLoading}
+      />
+
+      {(isEnsAddressLoading || isEnsNameLoading) && (
+        <p>Resolving...</p>
+      )}
+
+      {ensAddress && (
+        <p><strong>Resolved Address:</strong> {ensAddress}</p>
+      )}
+
+      {ensName && (
+        <div style={{ display: "flex", alignItems: "center", gap: "0.5rem" }}>
+          {ensAvatar && <img src={ensAvatar} alt={ensName} style={{ width: "24px", height: "24px", borderRadius: "50%" }} />}
+          <span><strong>Resolved ENS:</strong> {ensName}</span>
+        </div>
+      )}
+    </div>
+  );
+}
+```
+
+### With Error Handling
+
+<DocsProvider>
+  <UseAddressInputErrorExample />
+</DocsProvider>
+
+```tsx
+import React, { useState } from "react";
+import { useAddressInput } from "@scaffold-ui/hooks";
+
+function AddressInputWithErrors() {
+  const [value, setValue] = useState("");
+  const {
+    ensAddress,
+    isEnsAddressLoading,
+    isEnsAddressError,
+    isEnsAddressSuccess
+  } = useAddressInput({ value });
+
+  const showError = !isEnsAddressLoading && value && (isEnsAddressError || (isEnsAddressSuccess && ensAddress === null));
```

Comment:
for example here and in some other places you're checking for `null`, see https://github.com/scaffold-eth/scaffold-ui/pull/53/files#r2466132712